lib/modules/manager/cargo/artifacts.ts

```diff
@@ -64,11 +82,25 @@ export async function updateArtifacts({
   try {
     await writeLocalFile(packageFileName, newPackageFileContent);
     logger.debug('Updating ' + lockFileName);
-    await cargoUpdate(
-      packageFileName,
-      isLockFileMaintenance,
-      config.constraints?.rust
-    );
+
+    if (isLockFileMaintenance) {
+      await cargoUpdate(packageFileName, true, config.constraints?.rust)
+    } else if (updatedDeps.find(dep => !dep.currentVersion || !dep.newVersion)) {
+      await cargoUpdate(packageFileName, false, config.constraints?.rust)
```

Comment:
Can se assume `currentVersion` and `newVersion` are always there? The only reason I handle this is because exists unit tests don't include them. I guess `currentVersion` could be missing or incorrect if there is no `Cargo.lock` file present.

Reply:
If a lock file is present then the `currentVersion` should have been derived from it (via `lockedVersion`) and then you can assume you find it in the lock file. However it's also good to code defensively anytime it is easy to do!

Reply:
Yeah I guess the only cases where it wouldn't be present is if the lock file was missing, or the lock file and package files were desynced. I'm not really sure what the right approach is if we have a mixture of deps where some have `currentVersion` and some don't.

What we could do:
1. Update the entire lock file
2. Update just the deps that have `currentVersion` and ignore the rest
3. Bail, as this most likely means that the package and lock files are desynced

Reply:
If the lock file is missing then you won't get to this "artifacts updating" logic